src/hooks/useRovingCellRef.ts

```diff
@@ -1,33 +1,41 @@
+import type { MutableRefObject } from 'react';
 import { useCallback, useState } from 'react';
 import { scrollIntoView } from '../utils';
 
 // https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex
-export function useRovingCellRef(isSelected: boolean) {
+export function useRovingCellRef(isSelected: boolean, skipCellFocus?: MutableRefObject<boolean>) {
   // https://www.w3.org/TR/wai-aria-practices-1.1/#gridNav_focus
   const [isChildFocused, setIsChildFocused] = useState(false);
 
   if (isChildFocused && !isSelected) {
     setIsChildFocused(false);
   }
 
-  const ref = useCallback((cell: HTMLDivElement | null) => {
-    if (cell === null) return;
-    scrollIntoView(cell);
-    if (cell.contains(document.activeElement)) return;
-    cell.focus({ preventScroll: true });
-  }, []);
+  const ref = useCallback(
+    (cell: HTMLDivElement | null) => {
+      if (cell === null) return;
+      scrollIntoView(cell);
+      if (cell.contains(document.activeElement)) return;
+      if (skipCellFocus?.current) {
+        skipCellFocus.current = false;
```

Comment:
Not sure if this is safe 🤔

Reply:
It is safe.
https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node